sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/http/rest/SwaggerMethodParser.java

```diff
@@ -384,11 +383,11 @@ public void setHeaders(Object[] swaggerMethodArguments, HttpHeaders headers, Obj
                 final Object methodArgument = swaggerMethodArguments[headerSubstitution.getMethodParameterIndex()];
 
                 if (methodArgument instanceof Map) {
-                    @SuppressWarnings("unchecked") final Map<String, ?> headerCollection =
-                        (Map<String, ?>) methodArgument;
+                    @SuppressWarnings("unchecked") final Map<HttpHeaderName, ?> headerCollection =
```

Comment:
@srnagar should we change code generation to use `HttpHeaders` in this case?

If that isn't possible, I think this will need to remain as `Map<String, ?>`

Reply:
Let's keep this `Map<String, ?>` for now as this would require changes in multiple layers to switch to HttpHeaders in codegen.